spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/RedisAutoConfiguration.java

```diff
@@ -214,6 +214,10 @@ private JedisPoolConfig jedisPoolConfig() {
 			config.setMaxIdle(props.getMaxIdle());
 			config.setMinIdle(props.getMinIdle());
 			config.setMaxWaitMillis(props.getMaxWait());
+			config.setTestOnCreate(props.isTestOnCreate());
+			config.setTestOnBorrow(props.isTestOnBorrow());
```

Comment:
It would make sense to keep `testOnBorrow` `true` because you don't want to get broken connections from the pool.

For all other properties, it would make more sense to accept a `JedisPoolConfig` bean, if a bean is configured externally. `testOnCreate`/`testWhileIdle`/`testOnCreate` are very specific and not required to set by the majority of users. Pooling config can be sometimes very specific so it makes no sense to include the more advanced properties as it increases mostly complexity.